mypy deduced type on Request.user_agent_class is Type[_UserAgent]

I ran into this after updating to v2.0.2 of flask and werkzeug.

Since `UserAgent` is going away, I replaced it with `ua_parser` (example will follow).
However, setting `user_agent_class = MyUserAgent` triggers a typing error as the type of `Request.user_agent_class` refers to the internal (deprecated) `_UserAgent`.

Absolutely minimal reproduction:
```python
# minimal.py
from flask import Request
from werkzeug.user_agent import UserAgent

class MyUserAgent(UserAgent): pass

class MyRequest(Request):
    user_agent_class = MyUserAgent
```
produces
```shell
$ mypy minimal.py
minimal.py:7: error: Incompatible types in assignment (expression has type "Type[MyUserAgent]", base class "Request" defined the type as "Type[_UserAgent]")
Found 1 error in 1 file (checked 1 source file)
```

I've also created [a simple reproduction of the failing mypy run in a repo](https://github.com/sirosen/repro/tree/main/flask/ua_typesig), which includes both the above minimal example, and my usage with `ua_parser`.

I know that `_UserAgent` is going away, so maybe this doesn't matter. But I could submit a patch to change [this line](https://github.com/pallets/werkzeug/blob/f7b53ee8a81f015c6c85659959004328237a3c8c/src/werkzeug/sansio/request.py#L98) to read
```python
    user_agent_class: t.Type[UserAgent] = t.cast(t.Type[UserAgent], _DeprecatedUserAgent)
```

I think that would solve this, since the type would be known to `mypy` (and other type checkers) as `Type[UserAgent]`, and inheriting subclasses would be accepted.

Would that fix be welcome?

Also, I haven't seen any python projects establish a testing process for type annotations. Do the pallets projects have an approach if I wanted to add a test case for this?

--- a/CHANGES.rst
+++ b/CHANGES.rst
@@ -10,6 +10,7 @@ Unreleased
     ``HTTPException.get_response``, and ``Map.bind_to_environ`` accepts
     ``Request`` in addition to ``WSGIEnvironment`` for the first
     parameter. :pr:`2290`
+-   Fix type annotation for ``Request.user_agent_class``. :issue:`2273`
 
 
 Version 2.0.2

--- a/src/werkzeug/sansio/request.py
+++ b/src/werkzeug/sansio/request.py
@@ -95,7 +95,7 @@ class Request:
     #: .. versionadded:: 0.6
     list_storage_class: t.Type[t.List] = ImmutableList
 
-    user_agent_class = _DeprecatedUserAgent
+    user_agent_class: t.Type[UserAgent] = _DeprecatedUserAgent
     """The class used and returned by the :attr:`user_agent` property to
     parse the header. Defaults to
     :class:`~werkzeug.user_agent.UserAgent`, which does no parsing. An

